wgpu-core/src/device/mod.rs

```diff
@@ -294,14 +294,18 @@ impl DeviceLostClosure {
     ///
     /// - Both pointers must point to `'static` data, as the callback may happen at
     ///   an unspecified time.
-    pub unsafe fn from_c(closure: DeviceLostClosureC) -> Self {
+    pub unsafe fn from_c(mut closure: DeviceLostClosureC) -> Self {
         // Build an inner with the values from closure, ensuring that
         // inner.called is false.
         let inner = DeviceLostClosureC {
             callback: closure.callback,
             user_data: closure.user_data,
             called: false,
         };
+
+        // Mark the original closure as called, so we can safely drop it.
+        closure.called = true;
```

Comment:
This doesn't actually do anything - `closure` is dropped immediately after this.

Reply:
It prevents the `drop` impl from panicking (it asserts that we don't leak the closure by checking `called`).

Reply:
I've changed the name of the member from "called" to "consumed" to clarify its purpose, and why it is mutated in this case.